text/0000-uninstall-node.md

```diff
@@ -0,0 +1,64 @@
+- Feature Name: uninstall_node
+- Start Date: 2024-08-21
+- RFC PR:[https://github.com/volta-cli/volta/pull/1882]
+- Volta Issue:
+
+# Summary
+
+[summary]: #summary
+
+Add support for uninstalling node to `volta uninstall`
+
+# Motivation
+
+[motivation]: #motivation
+
+Currently, Volta doesn't support uninstalling node due to this main reasons(as discussed in [#327](!https://github.com/volta-cli/volta/issues/327)): Its main purpose is to save disk space, nowadays disk size is not a big problem in many cases. So this task is in a low priority.But community has this demand since 2019, they need to manually remove some files to achieve that.Add support for uninstalling node is more convinent in this case, btw since we have `install` here we should also have `uninstall` here.
+
+# Pedagogy
+
+[pedagogy]: #pedagogy
+
+The user's intuition and way of treating it are opposite to `install`. Since `install` fetch node image, unpack it and set it as default, `unintall` should remove those things.
+
+# Details
+
+[details]: #details
+
+The new command will have the structure:
+
+```
+volta uninstall node@version
```

Comment:
At that time I think ```uninstall``` should receive totally same args with ```install``` and treat them the same way,That means:
1. We don't requre a full version. Just as logic in ```install```, we support no verison, semver version , exact version and tag version.(Of course when ```node_index_file``` expires and fetch the new version, those none version, semver version and tag version will be resolved to a different one with the one when ```install``` runs. But I don't think it's a big deal)     

But I agree with your second thought, that feels like a better user experience indeed.

Reply:
I agree that “just support the same argument variants as `install`” makes a lot of sense. One note, though: it does feel like the kind of thing that might warrant a prompt, which we have yet to do anywhere. “This will remove [list all affected versions]; do you want to proceed?” It *might* be fine not to, but that could be a shockingly wide-ranging effect. 🤔

Reply:
I totally agree with you, a kind prompt like this is really important for users. You are soooo good at user experience.